### PR TITLE
Integration of ZDC fastsim prototype into O2.

### PR DESCRIPTION
**Pull request by SwirtaB**
Prototype provides interface for running two ML models in O2. CPU only for now.

Had to revert commit dd8e8f134041d3b8c5875042a8c9bf1266c39a14 due to compile error.

Also it is required to add _ONNXRuntime_ entry in o2.sh in alidist.

### CODE REVIEW

File: Utilities/ShmManager/src/ShmManager.cxx
Comment (sawenzel) [inline Utilities/ShmManager/src/ShmManager.cxx]: Please don't change this file. It has nothing to do with your feature and just taking it out is not a solution.
If you have problem with this file, please open a JIRA ticket describing the issue.

**Review by sawenzel — DISMISSED** [review]
Approving this to trigger CI testing once. Please don't merge yet.

File: Detectors/ZDC/fastsimulation/include/FastSimulations.hpp
Comment (sawenzel) [inline Detectors/ZDC/fastsimulation/include/FastSimulations.hpp]: try to use `const &` consistently in all interfaces taking complex objects

**Comment by sawenzel** [comment]
Thanks. Let's see how it compiles. 

A couple of preliminary comments:
a) please add copyright notices to all new files
b) please format your code with git-clang-format
c) please follow conventions for filenames and suffices (we typically use .h instead of .hpp and .cxx for source)
d) please follow conventions for variable names (member variables are `mFooBar`) and function names (the main Readme contains links to coding conventions)
e) please don't commit binary files. They need to be distributed by other means (we have CVMFS for this)
f) please put the stuff into `o2::` namespace (better `o2::zdc::fastsim`)

File: Utilities/ShmManager/src/ShmManager.cxx
Comment (SwirtaB) [reply Utilities/ShmManager/src/ShmManager.cxx]: Sorry for that. I revoked those changes in newest commits.

File: Detectors/ZDC/fastsimulation/include/FastSimulations.hpp
Comment (SwirtaB) [reply Detectors/ZDC/fastsimulation/include/FastSimulations.hpp]: Fixed in newest commits.

**Review by sawenzel — DISMISSED** [review]

**Comment by SwirtaB** [comment]
Added conditional build in CMakeLists.txt. ONNXRuntime is no longer required to build O2 (fast simulation module won't build).

**Review by sawenzel — DISMISSED** [review]

**Review by sawenzel — DISMISSED** [review]

**Comment by sawenzel** [comment]
@SwirtaB : I think we are getting there. The only remaining issue is to fix the full-CI test. It complains with:
```
## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/FastSimulations.cxx:46:13: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/FastSimulations.cxx:139:23: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/FastSimulations.cxx:143:25: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/FastSimulations.cxx:147:26: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/FastSimulations.cxx:209:23: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/FastSimulations.cxx:213:25: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/FastSimulations.cxx:217:26: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/Utils.cxx:50:22: error: statement should be inside braces [readability-braces-around-statements]
--
```

**Comment by SwirtaB** [comment]
I have fixed those errors, should be fine now.

**Review by sawenzel — APPROVED** [review]

**Pull request merged**
